DevoxxClientMobile/src/main/java/com/devoxx/views/SessionsPresenter.java

```diff
@@ -260,7 +260,13 @@ private Node createSessionsList(ContentDisplayMode mode) {
             Comparator<Session> sessionComparator = (s1, s2) -> {
                 int compareStartDate = s1.getStartDate().compareTo(s2.getStartDate());
                 if (compareStartDate == 0) {
-                    return s1.getEndDate().compareTo(s2.getEndDate());
+                    int compareEndDate = s1.getEndDate().compareTo(s2.getEndDate());
+                    if (compareEndDate == 0) {
+                        if (s1.getTalk() != null && s2.getTalk() != null) {
```

Comment:
shouldn't this get higher priority than the enddate? It's just a matter of definitions, but in case 2 talks start at the same moment, the next filter criterium might rather be type instead of enddate?

Reply:
As is right now, we group all the talks that have the same slot (same start-end date), and then with my proposed change we'll group them by type of talk (showing it with a label on top of the first one of such type and group). This solves the problem of showing the type of talk several times, even if slots are the same.

About the sort criterium, for instance, let's say we have lab sessions of 1 hour, and others of 3 hours, and also at the same starting time, conference sessions of one hour.

With the PR as is, the order would be conference sessions + lab sessions of 1 hour + lab sessions of 3 hours (three blocks, three labels).

But with your suggestion the order would be: conference sessions + lab sessions (two blocks but with sessions of different duration, grouped by same end date), so there is a small difference, but nothing significative.

Also, not all sessions have talks (i.e. coffee breaks), so moving up the type of talk criterium makes the comparator more complex.